crates/node-bindings/src/parcel/tracer.rs

```diff
@@ -0,0 +1,178 @@
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+
+use napi::bindgen_prelude::FromNapiValue;
+use napi::bindgen_prelude::ToNapiValue;
+use napi::JsNumber;
+use napi::JsObject;
+use napi::JsString;
+use once_cell::sync::Lazy;
+use parking_lot::Mutex;
+use tracing_appender::non_blocking::WorkerGuard;
+
+/// Ensures there is only ever one tracer running, initialized the first time it is needed.
+/// When it's dropped certain resources related to tracing might flush to disk or be released.
+/// It should be kept alive for the duration of the program or the Parcel instance.
+static GLOBAL_TRACER: Lazy<Mutex<Option<Tracer>>> = Lazy::new(|| Default::default());
+
+#[derive(Debug)]
+pub enum TracerMode {
+  /// Disable the Tracer
+  Disabled,
+  /// Output the Tracer logs to Stdout
+  Stdout,
+  /// Output the Tracer logs to a file
+  File {
+    /// The directory where the log files will be written.
+    directory: String,
+    /// A prefix for the log file names.
+    prefix: String,
+    /// The maximum number of rotated files to keep.
+    max_files: u32,
+  },
+}
+
+/// Telling Napi how to convert from a JS type to the enum
+impl FromNapiValue for TracerMode {
```

Comment:
Can't you just annotate the enum with napi since it's already in node bindings?

Reply:
Unfortunately structured enums are not supported by napi-rs and need to be manually mapped. An enum with internal values also has weird mappings.

I'm thinking of writing a macro to standardize enum serialization/deserialization so that an enum always serializes to/from `{ key: string, ...data: EnumProps }`

Reply:
Updated